### PR TITLE
[#960] [Fix] Make mnist application runnable from install

### LINKED ISSUE #731: Save path has a default value
Currently save_path has a default value when set from ini but we don't have default save_path when setting from a code.

```cpp
  model.save_path = iniparser_getstring(ini, "Model:Save_path", "./model.bin");
```

We have two options

1. moving default save_path to `Neuralnet::Constructor()`
2. remove default save_path

### PR DESCRIPTION
**Pull request by zhoonit**
commit 1 - [App] install application 
```
mnist_trainingSet.dat was missing in the installation,
which prevented the app to run.

This patch fixes the issue.

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

See #893
```

commit 2 - [Fix/Bug] NeuralNet check default save path
```
Add a check to see if save_path is valid if not, issue a warning

**Semantic changes:**
From this patch, default save path is gone

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>
```

resolves #731

### CODE REVIEW

**Comment by zhoonit** [comment]
Please note that this patch requires save path modification to pass the test, it will be done and this PR will be rebased.

File: nntrainer/models/model_loader.cpp [hunk 1]
```diff
@@ -62,8 +62,8 @@ int ModelLoader::loadModelConfigIni(dictionary *ini, NeuralNetwork &model) {
   model.loss_type = (LossType)parseType(
     iniparser_getstring(ini, "Model:Loss", unknown), TOKEN_LOSS);
   const std::string &save_path =
-    iniparser_getstring(ini, "Model:Save_path", "./model.bin");
-  model.setSavePath(model.app_context.getWorkingPath(save_path));
+    iniparser_getstring(ini, "Model:Save_path", "tmp_model.bin");
+  model.setSavePath(save_path);
```
Comment (kparichay) [inline nntrainer/models/model_loader.cpp:66]: I wonder why the prefix of `getWorkingPath` has been removed?

Comment (zhoonit) [reply nntrainer/models/model_loader.cpp:66]: `model.setSavePath` internally calls `save_path` so this is not needed here.

**Comment by zhoonit** [comment]
This pr needs a patch regarding #731 before merge

File: nntrainer/models/model_loader.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/models/model_loader.cpp:66]: So, if model has a `getSavePath` later, it will also have to add `getWorkingPath`?

Comment (zhoonit) [reply nntrainer/models/model_loader.cpp:66]: `setSavePath` adds it so it won't need `getWorkingPath`

**Comment by zhoonit** [comment]
v2. amended last commit to remove default path to resolve #731 

Add a check to see if save_path is valid if not, issue a warning

**Semantic changes:**
From this patch, default save path is gone

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**